include/unifex/create.hpp

```diff
@@ -34,9 +53,18 @@ struct _op {
       : rec_((Receiver&&) rec), fn_((Fn&&) fn), ctx_((Context&&) ctx) {}
 
     template (typename... Ts)
-      (requires receiver_of<Receiver, Ts...>)
-    void set_value(Ts&&... ts) noexcept(is_nothrow_receiver_of_v<Receiver, Ts...>) {
-      unifex::set_value((Receiver&&) rec_, (Ts&&) ts...);
+      (requires receiver_of<Receiver, Ts...> AND (convertible_to<Ts, ValueTypes> && ...))
+    void set_value(Ts&&... ts) noexcept {
+      UNIFEX_TRY {
+        // Satisfy the value completion contract with _do_convert, which
+        // ensures the value passed to the receiver conforms to
+        // the value types of the create sender. For example, if set_value
+        // is called with an lvalue reference but create sends non-reference
+        // values, _do_convert decays the provided Ts.
+        unifex::set_value(std::move(rec_), _do_convert<Ts&&, ValueTypes>{}(static_cast<Ts&&>(ts))...);
```

Comment:
Now that we've agreed on an implementation of `_do_convert` and what type parameters to pass to it, is this any different?
```suggestion
        unifex::set_value(std::move(rec_), ValueTypes{static_cast<Ts&&>(ts)}...);
```

If it's the same behaviour, do you see value in the different spelling? I might be convinced that the `_do_convert` version makes the intention clearer.

Reply:
Ah yes, `static_cast<ValueTypes>(static_cast<Ts&&>(ts)...)` works but the `ValueTypes{...}`does not (I get `-Wc++11-narrowing` with the  int -> double conversion test, and the `create<A>` test fails since `set_value<B>` tries to call `A`'s aggregate initialization with a `B` which does not work (I think that's what is going on at least).  `B` is convertible to `A` via the `operator A() const` member of `B`.